contracts/token/erc20/erc20.go

```diff
@@ -0,0 +1,446 @@
+	}
+
+	dec, err := library.BytesToUint8(bytes)
+	if err != nil {
+		return 0, err
+	}
+
+	return dec, nil
+}
+
+// Mint creates new tokens and adds them to minter's account balance
+// This function triggers a Transfer event
+func (erc20 *ERC20) Mint(ctx context.ContextInterface, msg context.Message, to string, amount uint64) error {
+	var err error
+	toAddr := library.Address(to)
+
+	// Nonce Check & Increase
+	if err = erc20.INonce.Check(ctx, ctx.MsgSender().String(), msg.Nonce); err != nil {
+		return err
+	}
+	if _, err = erc20.INonce.Increment(ctx, ctx.MsgSender().String()); err != nil {
+		return err
+	}
+
+	// beforeTokenTransfer
+
+	balanceKey, err := ctx.GetStub().CreateCompositeKey(BalancePrefix, []string{to})
+	if err != nil {
+		return err
+	}
+	preVal, err := ctx.GetStub().GetState(balanceKey)
+	if err != nil {
+		return err
+	}
+	preBalance, err := library.BytesToUint64(preVal)
+	if err != nil {
+		return err
+	}
+	err = ctx.GetStub().PutState(balanceKey, []byte(library.Uint64ToString(amount+preBalance)))
+	if err != nil {
+		return err
+	}
+
+	if err = ctx.EmitEvent("TransferSingle", &EventTransfer{
+		Operator: ctx.MsgSender(),
+		From:     library.ZeroAddress,
+		To:       toAddr,
+		Value:    amount,
+	}); err != nil {
+		return errors.Wrap(err, "Event TransferSingle")
+	}
+
+	// afterTokenTransfer
+
+	return nil
+}
+
+// Burn redeems tokens the minter's account balance
+// This function triggers a Transfer event
+func (erc20 *ERC20) Burn(ctx context.ContextInterface, msg context.Message, amount uint64) error {
+	var err error
+	fromAddr := library.Address(ctx.MsgSender().String())
+
+	// Nonce Check & Increase
+	if err = erc20.INonce.Check(ctx, ctx.MsgSender().String(), msg.Nonce); err != nil {
+		return err
+	}
+	if _, err = erc20.INonce.Increment(ctx, ctx.MsgSender().String()); err != nil {
+		return err
+	}
+
+	// beforeTokenTransfer
+
+	balanceKey, err := ctx.GetStub().CreateCompositeKey(BalancePrefix, []string{ctx.MsgSender().String()})
+	if err != nil {
+		return err
+	}
+	preVal, err := ctx.GetStub().GetState(balanceKey)
+	if err != nil {
+		return err
+	}
+	preBalance, err := library.BytesToUint64(preVal)
+	if err != nil {
+		return err
+	}
+
+	if amount > preBalance {
+		return fmt.Errorf("burning more than it remain")
+	}
+
+	err = ctx.GetStub().PutState(balanceKey, []byte(library.Uint64ToString(preBalance-amount)))
+	if err != nil {
+		return err
+	}
+
+	if err = ctx.EmitEvent("Transfer", &EventTransfer{
+		Operator: ctx.MsgSender(),
+		From:     fromAddr,
+		To:       library.ZeroAddress,
+		Value:    amount,
+	}); err != nil {
+		return errors.Wrap(err, "Event TransferSingle")
+	}
+
+	// afterTokenTransfer
+
+	return nil
+}
+
+// Transfer transfers tokens from client account to recipient account.
+// This function triggers a Transfer event.
+func (erc20 *ERC20) Transfer(ctx context.ContextInterface, msg context.Message, from string, to string, amount uint64) error {
```

Comment:
![image](https://user-images.githubusercontent.com/30621793/231935395-3ca35810-e2c6-4825-a2b8-42feaefa20f4.png)

Transfer should only have two parameters 
- to
- amount